include/pybind11/detail/internals.h

```diff
@@ -273,7 +273,9 @@ PYBIND11_NOINLINE inline internals &get_internals() {
         auto *&internals_ptr = *internals_pp;
         internals_ptr = new internals();
 #if defined(WITH_THREAD)
-        PyEval_InitThreads();
+        #if PY_VERSION_HEX >= 0x03090000
```

Comment:
I accidentally first wrote `0x03080000` here and it broke the tests, even though the CPython docs say that `PyEval_InitThreads()` doesn't do anything ever since 3.7.

This only makes me less certain that this pull request actually works.